src/main/java/org/cryptomator/frontend/fuse/ReadOnlyDirectoryHandler.java

```diff
@@ -31,7 +32,12 @@ public ReadOnlyDirectoryHandler(FileAttributesUtil attrUtil) {
 	}
 
 	public int getattr(Path path, BasicFileAttributes attrs, FileStat stat) {
-		stat.st_mode.set(FileStat.S_IFDIR | 0555);
+		if (attrs instanceof PosixFileAttributes) {
```

Comment:
i try to avoid the usage of `instanceof`, but here it is used in acceptable way. 
The only other solution i see would be extending all getattr() methods with an aditional argument (something like Optional<Long> and already converting here the Set of PosixPermissions).

@overheadhunter what do you think?

Reply:
`instanceof` is fine in this case. First I thought that we should _always_ read the PosixFileAttributes and just assert the underlying filesystem supports it (since we're using FUSE only on Mac and Linux). But since a user might store a vault on a mounted FAT32 partition, it is probably best to distinguish Posix vs. Basic file attrs.